sycl/include/sycl/reduction.hpp

```diff
@@ -821,8 +821,9 @@ auto make_reduction(RedOutVar RedVar, RestTy &&...Rest) {
 namespace reduction {
 inline void finalizeHandler(handler &CGH) { CGH.finalize(); }
 template <class FunctorTy> void withAuxHandler(handler &CGH, FunctorTy Func) {
-  CGH.finalize();
+  auto E = CGH.finalize();
```

Comment:
`finalize` returns an `event`, right? If so I would prefer to just have the type explicit here.

Reply:
Done.